### PR TITLE
eosmetrics: Allow skipping submission when EOS_DISABLE_METRICS is set

### PR DESCRIPTION
**Pull request by pwithnall**
Add an EOS_DISABLE_METRICS environment variable which, when set, causes
metrics to be quietly *not* submitted to the metrics daemon, and for the
submission code to still return successfully.

This is intended to be used from within the unit tests of other modules,
to avoid subprocesses which use eos-metrics submitting spurious metrics
to the server, or emitting warnings due to the metrics daemon not
running (and hence aborting due to G_DEBUG=fatal-warnings).

Signed-off-by: Philip Withnall <withnall@endlessm.com>

https://phabricator.endlessm.com/T14934

### CODE REVIEW

File: eosmetrics/emtr-event-recorder.c [hunk 1]
```diff
@@ -304,6 +311,19 @@ send_events_to_dbus_finish_callback (EmerEventRecorderServer *dbus_proxy,
...
+ * skip submitting any metrics. This is intended to be set when running unit
+ * tests in other modules, for example, to avoid submitting metrics from unit
+ * test runs. */
+static gboolean
+disable_event_submission (void)
+{
+  const gchar *val = g_getenv ("EOS_DISABLE_METRICS");
+
+  return (val != NULL &&
+          (g_strcmp0 (val, "") == 0 || g_strcmp0 (val, "1") == 0));
```
Comment (ptomato) [inline eosmetrics/emtr-event-recorder.c]: If you've already done the NULL check then you can probably use regular `strcmp` here.

File: eosmetrics/emtr-event-recorder.c [hunk 2]
```diff
@@ -319,12 +339,22 @@ send_events_to_dbus (EmtrEventRecorder *self,
     emtr_event_recorder_get_instance_private (self);
 
   GVariantBuilder uuid_builder;
-  get_uuid_builder (parsed_event_id, &uuid_builder);
-  GVariant *event_id_variant = g_variant_builder_end (&uuid_builder);
+  GVariant *event_id_variant;
```
Comment (ptomato) [inline eosmetrics/emtr-event-recorder.c]: Why the change back to C89 declaration style?

Comment (pwithnall) [inline eosmetrics/emtr-event-recorder.c]: To avoid allocating the `GVariantBuilder` (and friends) and then immediately freeing them in the `disable_event_submission () == TRUE` case.

**Comment by pwithnall** [comment]
> As a separate comment, it would probably be fine to downgrade the warning to a `g_message`, since the whole point is that the client should fire and forget, and not care whether the metric made it to the server or not. I wouldn't remove it altogether though, since it's useful in journal logs to see if the event recorder daemon crashed.

I think it should remain as a `g_warning()`, since then it gets suitably highlighted in the journal — this is an unexpected failure which a developer would want to know about, since it means something’s wrong with the metrics daemon or D-Bus setup.

Agreed that it should not be removed altogether.
